### PR TITLE
feat(store): replace store with react-query for Environment sections

### PR DESCRIPTION
**Pull request by RemiBonnet**
# What does this PR do?

Add the Environment part with [React Query](https://react-query-v3.tanstack.com/)

- Improved performance: React Query optimizes data fetching by using caching, deduplication, and automatic background updates. This can lead to a significant improvement in performance and a better user experience.

- Simplified data fetching: React Query provides a simple and intuitive API for fetching data, managing loading and error states, and handling pagination and infinite scrolling. This can reduce the amount of code we need to write and make your code easier to maintain.

- Improved developer productivity: By providing a consistent and reliable way to fetch and manage data, React Query can improve developer productivity and reduce the time and effort required to implement complex data fetching logic.
---

## PR Checklist

- [x] This PR introduces breaking change(s) and has been labeled as such
- [x] This PR introduces new store changes
- [x] I have followed the library pattern i.e `feature`, `ui`, `data`, `utils`
- [x] I made sure the code is type safe (no any)

### CODE REVIEW

**Comment by evoxmusic** [comment]
A preview environment was automatically created via Qovery.
Click on the link below to follow its deployment and use it.
:point_right: [[PR] staging - feat(store): replace store with react-query for deployment stages section - 2023-03-06T16:16:11Z](https://new.console.qovery.com/organization/141c07c8-0dd9-4623-983b-3fdd61867255/project/977bd069-20f7-4b24-bd95-28b44174cf43/environment/f986caa4-8e73-4740-8d2f-38212b7fb843/services/general)

 Another comment will be posted when deployments are terminated

**Comment by RemiBonnet** [comment]
Your preview environment has been successfully deployed !
Click on the link below to open your service:
:point_right: [console](https://zf986caa4-z1ad61286-gtw.za8ad0659.bool.sh)
:point_right: [storybook](https://zf986caa4-zce771dbf-gtw.za8ad0659.bool.sh)

File: libs/domains/environment/src/lib/slices/deployment-stage.ts [hunk 1]
```diff
@@ -1,203 +1,212 @@
...
+) => {
+  const queryClient = useQueryClient()
+
+  return useMutation(
+    async ({ data }: { data: DeploymentStageRequest }) => {
+      const response = await deploymentStageMainCallApi.createEnvironmentDeploymentStage(environmentId, data)
+      return response.data
+    },
+    {
+      onMutate: async (variables) => {
```
Comment (RemiBonnet) [inline libs/domains/environment/src/lib/slices/deployment-stage.ts:118]: Not necessary to do it but :
- This function will fire before the mutation function is fired and is passed the same variables the mutation function would receive
- Useful to perform optimistic updates to a resource in hopes that the mutation succeeds
- The value returned from this function will be passed to both the onError and onSettled functions in the event of a mutation failure and can be useful for rolling back optimistic updates.

**Pull request merged**